guardian/prosemirror-elements
# feat: Resize interactive iframe based on content

## What does this change?
This PR causes the interactive element's preview to resize based on its content, provided that the interactive observes the `Window.postMessage` interface specified by [iframe-messenger](https://github.com/guardian/iframe-messenger).

Resizing the preview is more complicated for interactives than the existing embed, because interactives are on another domain, therefore CORS prevents us from directly querying the height of the iframe's window. Therefore, we can interact with the iframe's document only through the messaging API.

Along the way, we noticed that React elements representing our Prosemirror Elements were not being dismounted when we deleted or moved them. This meant that cleanup functions in the `useEffect` hook were never being called, and we were unable to cleanup, for example, event listeners that were registered in a Component. As a result, this PR also includes a fix to properly unmount elements.

## How to test

Run `prosemirror-elements` locally and add the interactive element. Does the height of the interactive element resize to fit the entire interactive? This may be easier to observe if you shrink the window, as the elements will grow to fullscreen width.

## Images


| Before | After |
| --- | --- |
|  ![2022-02-15 14 06 02](https://user-images.githubusercontent.com/34686302/154078304-bd7a858e-505d-4312-b53f-59e24a320caa.gif) | ![2022-02-15 14 05 20](https://user-images.githubusercontent.com/34686302/154078329-66c075e9-03c0-41e0-aa97-1eda58e061ce.gif) |

## Accessibility
